### PR TITLE
Don't close stdin when cleaning up after a --dry-run

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
This one was so much fun!

File: tar/multitape/multitape_write.c [hunk 1]
```diff
@@ -416,6 +416,9 @@ writetape_open(uint64_t machinenum, const char * cachedir,
 		goto err0;
 	memset(d, 0, sizeof(struct multitape_write_internal));
 
+	/* Initialize lockfd as "not locked". */
+	d->lockfd = -1;
```
Comment (gperciva) [inline tar/multitape/multitape_write.c:420]: I did consider omitting this part and changing the "check if ok to close" to:

```
if (d->lockfd > 0)
    close(d->lockfd);
```

but then I figured that if stdin was closed somewhere else (due to a new bug), then `multitape_lock(cachedir)` could return 0, which would then be interpreted as "not locked".  It seemed safer to go with -1, which is definitely not a real descriptor.

File: tar/multitape/multitape_write.c [hunk 2]
```diff
@@ -528,7 +531,8 @@ writetape_open(uint64_t machinenum, const char * cachedir,
 err5:
 	storage_write_free(d->S);
 err4:
-	close(d->lockfd);
+	if (d->lockfd >= 0)
```
Comment (cperciva) [inline tar/multitape/multitape_write.c:534]: `!= -1`.  There's only one value which we're using to indicate "cache directory is not locked".  Other negative values would be invalid but we shouldn't hide bugs like that.

**Comment by gperciva** [comment]
Rebased.

**Pull request merged**